xtask/src/main.rs

```diff
@@ -0,0 +1,364 @@
+use anyhow::{anyhow, Context, Result};
+use clap::{Parser, Subcommand};
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+
+#[derive(Parser)]
+#[command(name = "xtask")]
+#[command(about = "Development tasks for rs-matter ConnectedHomeIP integration")]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Setup ConnectedHomeIP environment for integration testing
+    ItestSetup {
+        /// ConnectedHomeIP repository reference (branch/tag/commit)
+        #[arg(long, default_value = "master")]
+        connectedhomeip_ref: String,
+        /// Force rebuild even if cached
+        #[arg(long)]
+        force_rebuild: bool,
+    },
+    /// Build rs-matter examples
+    Build {
+        /// Build profile (debug or release)
+        #[arg(long, default_value = "debug")]
+        profile: String,
+        /// Additional cargo features
+        #[arg(long)]
+        features: Vec<String>,
+    },
+    /// Run ConnectedHomeIP YAML integration tests
+    Itest {
+        /// Test names to run (if empty, runs all enabled tests)
+        tests: Vec<String>,
+        /// Build profile to use for rs-matter examples
+        #[arg(long, default_value = "debug")]
+        profile: String,
+        /// Timeout for each test in seconds
+        #[arg(long, default_value = "120")]
+        timeout: u64,
+        /// Skip building rs-matter (assume it's already built)
+        #[arg(long)]
+        skip_build: bool,
+    },
+}
+
+const DEFAULT_FEATURES: &[&str] = &["os", "rustcrypto", "log"];
+
+#[tokio::main]
```

Comment:
Do we really need `tokio` for a cmdline project? I don't think so. 

Please remove the `tokio` dependency and turn all async functions into sync. We are anyway not really using any async library or have any async code besides 3 functions in this file which actually do not even `.await` on anything, if I'm not mistaken.